core/trino-main/src/main/java/io/trino/cost/TableStatsProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import io.trino.Session;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.statistics.TableStatistics;
+
+public interface TableStatsProvider
+{
+    TableStatsProvider EMPTY = (session, tableHandle) -> TableStatistics.empty();
+
+    public TableStatistics getTableStatistics(Session session, TableHandle tableHandle);
```

Comment:
`public` is redundant in an interface. Did it cause a checkstyle failure? (it should, please check)